server/opts.go

```diff
@@ -1680,6 +1680,49 @@ func parseJetStreamForAccount(v interface{}, acc *Account, errors *[]error, warn
 	return nil
 }
 
+// takes in a storage size as either an int or a string and returns an int64 value based on the input.
+func getStorageSize(v interface{}) (int64, error) {
+	_, ok := v.(int64)
+	if ok {
+		return v.(int64), nil
+	}
+
+	s, ok := v.(string)
+	if !ok {
+		return 0, fmt.Errorf("must be int64 or string")
+	}
+
+	if s == "" {
+		return 0, nil
+	}
+
+	suffix := strings.ToUpper(s[len(s)-1:])
+	prefix := s[:len(s)-1]
+	var num int
```

Comment:
I would actually rewrite from here to the end something like that:
```
num, err := strconv.ParseInt("1", 10, 64)
if err != nil {
    return 0, err
}
suffixMap := map[string]int64{"K": 10, "M": 20, "G": 30, "T": 40}
mult, ok := suffixMap[suffix]
if !ok {
    return 0, fmt.Errorf("sizes defined as strings must end in K, M, G, T")
}
num *= 1 << mult
return num, nil
```
At any rate, even if we were to keep the switch, there is no need to parse the prefix in each "case" line, and also, we want to use the 64bit version so that `num` is not a 32bit if server is built as 32bit and would overflow with some of the units.

Reply:
Ah great points. Yeah let me redo that.